pkg/sql/distsqlrun/outbox.go

```diff
@@ -283,6 +283,9 @@ func (m *outbox) mainLoop(ctx context.Context) error {
 							return err
 						}
 					}
+				} else {
+					// Case in which we are not recording but still need to finish span.
+					tracing.FinishSpan(span)
```

Comment:
Can this be done in a `defer` instead, or is it not possible?

Reply:
We want to Finish it before calling `getTraceData` above.. But I think we could pull out the `FinishSpan` and `getTraceData` code out of the `if !ok {}`. If we're not recording, `getTraceData` would just return `nil`.

Reply:
Also, are we leaking the span in the `return nil` case above (line 273)?

Reply:
I think pulling it out (as well as the flush) would work. Good catch on line 273

Reply:
Done. Logging errors now (should they be info?).